letsencrypt-win-simple/Plugin/SanByDomainPlugin.cs

```diff
@@ -0,0 +1,288 @@
+
+                    curDom = item.DomainName;
+                }
+
+                Console.WriteLine();
+                Console.WriteLine("Enter Id of domain to make SAN");
+                var domainInput = Console.ReadLine();
+                int domId = -1;
+                try
+                {
+                    domId = Convert.ToInt32(domainInput);
+
+                    var selecteddomain = domains.FirstOrDefault(o => o.DomainId == domId);
+                    if (selecteddomain != null)
+                    {
+                        var sanDomains = domains.Where(o => o.DomainName == selecteddomain.DomainName).OrderBy(o => o.IsDomain == false).ThenBy(o => o.Host).ToList();
+
+                        if (sanDomains.Count > SanMax)
+                        {
+                            Console.WriteLine($" You have too many hosts for a SAN certificate. Let's Encrypt currently has a maximum of " + SanMax + " alternative names per certificate.");
+                            Log.Error("You have too many hosts for a San certificate. Let's Encrypt currently has a maximum of " + SanMax + " alternative names per certificate.");
+                        }
+                        else
+                        {
+                            List<Target> siteList = new List<Target>();
+                            List<string> altList = new List<string>();
+
+                            foreach (var item in sanDomains.OrderBy(o => o.IsDomain == false).ThenBy(o => o.Host))
+                            {
+                                Target target = new Target();
+                                target.Host = item.Host;
+                                target.WebRootPath = item.WebRootPath;
+                                target.SiteId = item.SiteId;
+                                siteList.Add(target);
+                            }
+
+                            Target totalTarget = CreateTarget(siteList);
+                            ProcessTotaltarget(totalTarget, siteList);
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("Invalid Id...");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("An error occured!");
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(ex.Message);
+                    Console.ResetColor();
+                }
+            }
+        }
+
+        public override void Auto(Target target)
+        {
+            Console.WriteLine("Auto isn't supported for SanByDomain Plugin");
+        }
+
+        public override void Renew(Target target)
+        {
+            Console.WriteLine("Renew isn't supported for SanByDomain Plugin");
+        }
+
+        //Mostly copied from IISSiteServerPlugin
+        private Target CreateTarget(List<Target> sites)
```

Comment:
This isent possible I just wrote this comment not to take credit for the whole part...